accepted/2020/workloads/workload-manifest.md

```diff
@@ -0,0 +1,430 @@
+
+## Installation
+
+Manifests are installed to `dotnet/sdk/{sdk-band}-manifests/{manifest-id}/` in an unpacked form. These are called the *installed manifests*.
+
+The dotnet SDK would be expected to include baseline versions of the installed manifests for all manifests that are known and supported. Although any components listed in this baseline manifest need to work with that SDK version, it does not need to be fully up-to-date: it exists primarily for listing workloads that are available to be installed, and for its ID to be known by the SDK when fetching updated versions from NuGet.
+
+Any workload management operation (install, uninstall, repair update) must be a transactional operation that also updates the installed manifests to the latest available versions for that SDK band.
+
+## Advertising
+
+The .NET tooling will automatically and opportunistically download updated versions of all manifests for the current SDK band and unpack them to `~/.dotnet/sdk-advertising/{sdk-band}/{manifest-id}/`. These user-local updated copies of the manifest are known as *advertising manifests.* The advertising manifests are used when listing workloads that are available to be installed, and for the build tooling to produce warnings that newer versions of installed components are available*.* They are **not** used in pack resolution.
+
+# Format
+
+Manifests are json files. Comments are supported, both `//` and `/* */` styles.
+
+The toplevel is a JSON object, containing the following keys:
+
+| Key | Type | Value | Required |
+|--|--|--|--|
+| `version` | int | The version of the manifest. Must match the version of the NuGet package that contains the manifest. | Yes |
+| `description` | string | Description of the content and/or purpose of the manifest. This is primarily for commenting and/or diagnostic purposes and is not expected to be surfaced in the UX. | No |
+| `workloads` | object | Workload definitions keyed by workload ID. | No |
+| `packs` | object | Pack definitions keyed by pack ID. | No |
+| `data` | object | Allows manifests to include arbitrary data without risk of conflict. | No |
+
+## Workload Definitions
+
+Workload definitions take the following form:
+
+| Key | Type | Value | Required |
+|--|--|--|--|
+| `abstract` | bool | If `true`, this workload can only be extended, and is never exposed directly as an installable workload. Default is `false`. | No |
+| `kind` | string | Either `build` or `dev`. Default is `dev`. | No |
+| `description` | int | User-visible description for the workload. | Yes if dev and non-abstract |
+| `packs` | string array | IDs of the packs that are included in the workload. | No |
+| `extends` | string array | IDs of workloads whose packs should be included in this workload. | No |
+| `platforms` | string array | Limits the workload and workloads that extend it to only be shown and installed on these host platforms. The strings are RIDs. | False |
```

Comment:
I'm not sure if we want to use RIDs for this or not.  They may be more complex than we need, and I think we'll need the RID graph when processing the workload manifest.

Reply:
Ironically I didn't use RIDs in my implementation and you fixed it? Thanks :D